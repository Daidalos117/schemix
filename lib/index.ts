import { PrismaSchema } from "@/modules/PrismaSchema";

import { PrismaDataSourceOptions } from "@/@types/prisma-datasource";
import {
  PrismaGeneratorOptions,
  PrismaMultiGeneratorOptions,
} from "@/@types/prisma-generator";

interface CreateSchemaOptions {
  datasource: PrismaDataSourceOptions;
  generator: PrismaGeneratorOptions | PrismaMultiGeneratorOptions;
}

/**
 * Create a Prisma schema object.
 * @param props.datasource The datasource object information for Prisma.
 * @param props.generator The generator object information for Prisma.
 * @returns The generated schema object.
 */
export const createSchema = ({ datasource, generator }: CreateSchemaOptions) =>
  new PrismaSchema(datasource, generator);
