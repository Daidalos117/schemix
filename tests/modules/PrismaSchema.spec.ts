import { PrismaSchema } from "schemix/lib/index";
import { describe, expect, it } from "vitest";

describe("PrismaSchema", () => {
  it("Should expect multiple generators", async () => {
    const schema = new PrismaSchema(
      {
        provider: "cockroachdb",
        url: { env: "DATABASE_URL" },
      },
      [
        {
          provider: "prisma-client-js",
          name: "client",
        },
        {
          provider: "other-client",
          name: "other",
        },
      ]
    );

    const asString = await schema.toString();
    expect(asString).toMatchSnapshot();
  });
});
