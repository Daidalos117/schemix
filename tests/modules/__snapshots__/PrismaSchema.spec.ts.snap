// Vitest Snapshot v1

exports[`PrismaSchema > Should expect multiple generators 1`] = `
"datasource database {
  provider = \\"cockroachdb\\"
  url      = env(\\"DATABASE_URL\\")
}

generator client {
  provider = \\"prisma-client-js\\"
}

generator other {
  provider = \\"other-client\\"
}
"
`;
